File: mars-rover/typescript/tests/commandBuilder.ts
export class CommandBuilder {
  private buffer = '';

  forward(): this  { this.buffer += 'F'; return this; }
  backward(): this { this.buffer += 'B'; return this; }
  left(): this     { this.buffer += 'L'; return this; }
  right(): this    { this.buffer += 'R'; return this; }

  build(): string { return this.buffer; }
}

File: mars-rover/python/tests/command_builder.py
class CommandBuilder:
    def __init__(self) -> None:
        self._buffer = ""

    def forward(self) -> "CommandBuilder":
        self._buffer += "F"
        return self

    def backward(self) -> "CommandBuilder":
        self._buffer += "B"
        return self

    def left(self) -> "CommandBuilder":
        self._buffer += "L"
        return self

    def right(self) -> "CommandBuilder":
        self._buffer += "R"
        return self

    def build(self) -> str:
        return self._buffer
